builtin/providers/aws/resource_aws_elasticache.go

```diff
@@ -0,0 +1,245 @@
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set: func(v interface{}) int {
+					return hashcode.String(v.(string))
+				},
+			},
+			"security_group_ids": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set: func(v interface{}) int {
+					return hashcode.String(v.(string))
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsElasticacheCreate(d *schema.ResourceData, meta interface{}) error {
+	elasticacheconn := meta.(*AWSClient).elasticacheconn
+
+	clusterId := d.Get("cluster_id").(string)
+	nodeType := d.Get("node_type").(string)           // e.g) cache.m1.small
+	numNodes := int64(d.Get("num_cache_nodes").(int)) // 2
+	engine := d.Get("engine").(string)                // memcached
+	engineVersion := d.Get("engine_version").(string) // 1.4.14
+	port := int64(d.Get("port").(int))                // 11211
+	subnetGroupName := d.Get("subnet_group_name").(string)
+	securityNameSet := d.Get("security_group_names").(*schema.Set)
+	securityIdSet := d.Get("security_group_ids").(*schema.Set)
+	paramGroupName := d.Get("parameter_group_name").(string) // default.memcached1.4
+
+	securityNames := expandStringList(securityNameSet.List())
+	securityIds := expandStringList(securityIdSet.List())
+
+	req := &elasticache.CreateCacheClusterInput{
+		CacheClusterID:          aws.String(clusterId),
+		CacheNodeType:           aws.String(nodeType),
+		NumCacheNodes:           aws.Long(numNodes),
+		Engine:                  aws.String(engine),
+		EngineVersion:           aws.String(engineVersion),
+		Port:                    aws.Long(port),
+		CacheSubnetGroupName:    aws.String(subnetGroupName),
+		CacheSecurityGroupNames: securityNames,
+		SecurityGroupIDs:        securityIds,
+		CacheParameterGroupName: aws.String(paramGroupName),
+	}
+
+	_, err := elasticacheconn.CreateCacheCluster(req)
+	if err != nil {
+		return fmt.Errorf("Error creating Elasticache: %s", err)
+	}
+
+	// AWS CLI complains like below if we try to delete while creating.
+	//     Can only delete cache clusters with state in:
+	//     available, failed, incompatible-parameters, incompatible-network, restore-failed
+	// So we cannot delete clusters in `creating` state.
+	// We need to wait for states which we can delete.
+	pending := []string{"creating"}
+	stateConf := &resource.StateChangeConf{
+		Pending:    pending,
+		Target:     "available",
+		Refresh:    CacheClusterStateRefreshFunc(elasticacheconn, d.Id(), "available", pending),
+		Timeout:    10 * time.Minute,
+		Delay:      10 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	log.Printf("[DEBUG] Waiting for state to become available: %v", d.Id())
+	_, sterr := stateConf.WaitForState()
+	if sterr != nil {
+		return fmt.Errorf("Error waiting for elasticache (%s) to be deletable: %s", d.Id(), sterr)
```

Comment:
This message needs an update now that this lives in `Create` "to be deletable" -> "to be created"